joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/GoSrc2Cpg.scala

```diff
@@ -36,7 +36,9 @@ class GoSrc2Cpg extends X2CpgFrontend[Config] {
           astGenResult.parsedModFile.flatMap(modFile => GoAstJsonParser.readModFile(Paths.get(modFile)).map(x => x))
         )
         if (config.fetchDependencies) {
+          goGlobal.processingDependencies = true
```

Comment:
A suggestion in general, but if `goGlobal` was a case class, you could simply pass a deep copy as follows:
```scala
new DownloadDependenciesPass(goMod, goGlobal.copy(processingDependencies = true)).process()
```

Reply:
Earlier goGlobal was a singleton object converted to a class object in order to not clear the state. If I pass the copy of the same, the cache built in this pass will not be useful in further passes.

Reply:
OK I see, makes sense.